xclim/core/dataflags.py

```diff
@@ -0,0 +1,622 @@
+    extreme_high = _sanitize_attrs(da > thresh_converted)
+    extreme_high.attrs[
+        "description"
+    ] = f"Temperatures found in excess of {thresh} in {da.name}."
+    return extreme_high
+
+
+@_register_methods
+def negative_accumulation_values(
+    da: xarray.DataArray,
+) -> xarray.DataArray:
+    """Check if variable values are negative for any given day.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+
+    Returns
+    -------
+    xarray.DataArray, [bool]
+
+    Examples
+    --------
+    To gain access to the flag_array:
+
+    >>> ds = xr.open_dataset(path_to_pr_file)
+    >>> flagged = (ds.pr < 0)
+    """
+    negative_accumulations = _sanitize_attrs(da < 0)
+    negative_accumulations.attrs[
+        "description"
+    ] = f"Negative values found for {da.name}."
+    return negative_accumulations
+
+
+@_register_methods
+@declare_units(da="[precipitation]", check_output=False)
+def very_large_precipitation_events(
+    da: xarray.DataArray, thresh="300 mm d-1"
+) -> xarray.DataArray:
+    """Check if precipitation values exceed 300 mm/day for any given day.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+    thresh : str
+
+    Returns
+    -------
+    xarray.DataArray, [bool]
+
+    Examples
+    --------
+    To gain access to the flag_array:
+
+    >>> from xclim.core.units import convert_units_to
+    >>> ds = xr.open_dataset(path_to_pr_file)
+    >>> threshold = convert_units_to("300 mm d-1", ds.pr)
+    >>> flagged = (ds.pr > threshold)
+    """
+    thresh_converted = convert_units_to(thresh, da)
+    very_large_events = _sanitize_attrs(da > thresh_converted)
+    very_large_events.attrs[
+        "description"
+    ] = f"Precipitation events in excess of {thresh} for {da.name}."
+    return very_large_events
+
+
+@_register_methods
+@declare_units(da="[precipitation]", check_output=False)
+def many_1mm_repetitions(da: xarray.DataArray) -> xarray.DataArray:
+    """Check if precipitation values repeat at 1 mm/day for 10 or more days.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+
+    Returns
+    -------
+    xarray.DataArray, [bool]
+
+    Examples
+    --------
+    To gain access to the flag_array:
+
+    >>> from xclim.core.units import convert_units_to
+    >>> from xclim.indices.run_length import suspicious_run
+    >>> ds = xr.open_dataset(path_to_pr_file)
+    >>> threshold = convert_units_to("1 mm d-1", ds.pr)
+    >>> flagged = suspicious_run(ds.pr, window=10, op="==", thresh=threshold)
+    """
+    thresh = convert_units_to("1 mm d-1", da)
+    repetitions = _sanitize_attrs(suspicious_run(da, window=10, op="==", thresh=thresh))
+    repetitions.attrs[
+        "description"
+    ] = f"Repetitive precipitation values at 1mm d-1 for at least 10 days found for {da.name}."
+    return repetitions
+
+
+@_register_methods
+@declare_units(da="[precipitation]", check_output=False)
+def many_5mm_repetitions(da: xarray.DataArray) -> xarray.DataArray:
```

Comment:
Update on this: I was able to make a more general version of this flag, but due to the limitations in using yaml to trigger these checks, _I can't run the same check twice using different parameters_. I'm not sure how else this can be done. 

This will need to be revisited later.

Reply:
Is it due to how we declare flags in the yaml? Instead of a dict, we could have the section as a list?
```
data_flags:
    - negative_accumulation_values:
    - very_large_precipitation_events:
    - many_repetitions:
        thresh: 5 mm   
    - many_repetitions:
        thresh: 1 mm
```

In the code, this should appear as a list of dictionaries with a single key.

Reply:
If it's as simple as redefining these as lists, we can remove two indices from the dataflags. Nice!